### PR TITLE
[fix]: [CDS-85972]: adding exception context for helm

### PR DESCRIPTION
**Pull request by markoharness**
## Describe your changes
Created a TaskContext class to be use as a common context object during task execution.
Implementation is only for Helm task.

We are keeping infra connector info in the context object. Currently only Azure infra connector info has been implemented. For the rest of them can be done in subsequent PRs (seeing how this PR is already huge).

 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/57004)

### CODE REVIEW

**Comment by markoharness** [comment]
trigger uts

**Comment by markoharness** [comment]
trigger smartchecks

**Comment by markoharness** [comment]
trigger uts

**Comment by markoharness** [comment]
trigger uts

**Comment by markoharness** [comment]
trigger ut6

**Comment by markoharness** [comment]
trigger messagecheck

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/k8s/RancherK8sInfraDelegateConfig.java [hunk 1]
```diff
@@ -15,19 +16,28 @@
...
 import lombok.Builder;
 import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
 
 @CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
 @Value
 @Builder
 @OwnedBy(CDP)
 @RecasterAlias("io.harness.delegate.task.k8s.RancherK8sInfraDelegateConfig")
+@Slf4j
```
Comment (GoelPratyush99) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/k8s/RancherK8sInfraDelegateConfig.java:32]: not using this import anywhere in this file or for other similar classes

**Comment by markoharness** [comment]
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/k8s/RancherK8sInfraDelegateConfig.java` line 32 at r3](https://reviewable.io/reviews/harness/harness-core/57004#gh-1434863377:-NmFv2bWCu2f9JRkagXG:b-hfpv0l) ([raw file](https://github.com/harness/harness-core/blob/2dfe6c36b9bb7a2a73591e9a5a214c9f2a45a384/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/k8s/RancherK8sInfraDelegateConfig.java#L32)):*
<details><summary><i>Previously, GoelPratyush99 (Pratyush Goel) wrote…</i></summary><blockquote>

not using this import anywhere in this file or for other similar classes
</blockquote></details>

removed

**Comment by markoharness** [comment]
trigger uts

**Comment by markoharness** [comment]
trigger ut8

**Pull request closed without merging**